geo-traits/src/line_string.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{Dimension, LineStringIterator, PointTrait};
+use geo_types::{Coord, CoordNum, LineString};
+
+/// A trait for accessing data from a generic LineString.
+pub trait LineStringTrait: Sized {
+    /// The coordinate type of this geometry
+    type T: CoordNum;
+
+    /// The type of each underlying coordinate, which implements [PointTrait]
+    type ItemType<'a>: 'a + PointTrait<T = Self::T>
```

Comment:
Hmm, I agree this sounds nice, but I'll have to figure out how to update the iterator macros because `ItemType` is hard coded in there: https://github.com/georust/geo/blob/449b39fed135b8048ddd4129d1026902396661c4/geo-traits/src/iterator.rs#L37-L38

Reply:
Done in [`33c1ef9` (#1157)](https://github.com/georust/geo/pull/1157/commits/33c1ef99e50ac3b601ba93460d19d345b52d7602)